sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// A format used to specify how a model should be serialized and deserialized.
+    /// </summary>
+    public readonly partial struct ModelSerializerFormat : IEquatable<ModelSerializerFormat>
+    {
+        internal const string JsonValue = "J";
+        internal const string WireValue = "W";
+
+        /// <summary>
+        /// Default format which will serialize all properties including read-only and additional properties.
+        /// The format will always be JSON.
+        /// </summary>
+        public static readonly ModelSerializerFormat Json = new ModelSerializerFormat(JsonValue);
+
+        /// <summary>
+        /// Format used to serialize this model when sending as a request to an Azure service.
+        /// It will not serialize read-only properties or additional properties.
+        /// The content-type will vary between JSON, XML, etc., depending on the service.
+        ///
+        /// Most use cases should prefer a more complete format like <see cref="ModelSerializerFormat.Json"/> that includes
+        /// read-only and additional properties.
+        /// </summary>
+        public static readonly ModelSerializerFormat Wire = new ModelSerializerFormat(WireValue);
```

Comment:
What format would we expect users to pass for AMQP models that we want to implement the IModelSerializable interface? Wire doesn't really make sense as per the description, and JSON is not possible either. It is more like Wire vs Native format, where the Native format could vary based on the actual format in use.

Reply:
Wire is the native service format.  So if amqp models implement the interface they would pass in `W` but you could also opt in to implementing the `J` format and let people save the model as wire.  If you use anything other than wire you are expected to return all values including additional and readonly.  I changed the description to say that the wire `may` not serialize those since you are correct some wire formats do.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/38150